### ISSUE TITLE
Mulitple validation of a single property

### ISSUE THREAD

**Issue by k-achinger** [JeremySkinner/FluentValidation#471]
Hi,

I'm using FluentValidation in my WebAPI service and after upgrading the packages from 7.0.0-beta2 to 7.0.0, a following problem arised.

Take a look at a simplified scenario:
```cs

class A
{ }

class B
{
   public A A { get; set; }
}

class AValidator : AbstractValidator<A>
{ }

class BValidator : AbstractValidator<B>
{
   BValidator ()
   {
      this.RuleFor(o => o.A).SetValidator(new AValidator());
   }
}
```
Both validators are registered in the Unity container (instance of A can also be a controller method parameter):

```cs
container.RegisterType<IValidator<A>, AValidator>();
container.RegisterType<IValidator<B>, BValidator>();
```

Validating B results in executing the AValidator twice and giving duplicated validation results.

Any ideas? Thanks.

**Comment by JeremySkinner** [comment]
So that was introduced by this change: https://github.com/JeremySkinner/FluentValidation/issues/365 and this pull request: https://github.com/JeremySkinner/FluentValidation/pull/437 

It actually brings FV's WebApi integration in-line with the standard behaviour in WebApi (and also MVC), where WebApi will independently resolve a validator and execute it for each property in the request. This means that you essentially have 2 validators defined for the same property:

1) WebApi sees a property of Type A, asks FV for an AValidator and executes it
2) WebApi sees a type B, asks FV for a BValidator and executes it, which in turn executes a type A validator

Basically you need to pick 1 approach - either register the validator with the container and rely on "transient" property validation, or remove it from the container and use SetValidator for explicit validation, but don't do both.

As this is a breaking change from previous versions I'll look at making this behaviour configurable in the next release so you can revert to the old behaviour if you need to.

**Comment by k-achinger** [comment]
I'm using different constructor of B when validating it as a member of A than validating B directly, so I need to find some workaround for this.

Thanks for the quick reply. Have a nice day!

**Comment by JeremySkinner** [comment]
I've pushed 7.0.1 to nuget which contains a new property on the FluentValidationModelValidatorProvider called DisableDiscoveryOfPropertyValidators. If you set this to true then you'll get the 6.4 behaviour back. Please give it a try.

**Comment by k-achinger** [comment]
Works perfectly.
Thanks a lot 👍

**Comment by JeremySkinner** [comment]
Great, glad it's working!
